### PR TITLE
livingpickupevent

### PR DESCRIPTION
**Pull request by Faithcaio**
To expand on
https://github.com/SpongePowered/SpongeCommon/pull/1602
This will add Pickup Events for other Entities:
 - [x] EntityLiving
 - [x] Villager

### CODE REVIEW

File: src/main/java/org/spongepowered/common/mixin/core/entity/passive/MixinEntityVillager.java [hunk 1]
```diff
@@ -237,4 +248,25 @@ public void supplyVanillaManipulators(List<DataManipulator<?, ?>> manipulators)
     public Optional<Villager> getCarrier() {
         return Optional.of(this);
     }
+
+    @Redirect(method = "updateEquipmentIfNeeded",  at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/passive/EntityVillager;canVillagerPickupItem(Lnet/minecraft/item/Item;)Z"))
+    public boolean onCanVillagerPickUpItem(EntityVillager villager, Item item, EntityItem itemEntity) {
+        boolean result = this.canVillagerPickupItem(item);
+        if (!SpongeCommonEventFactory.callChangeInventoryPickupPreEvent(((EntityLiving)(Object) this), itemEntity)) {
```
Comment (JBYoshi) [inline src/main/java/org/spongepowered/common/mixin/core/entity/passive/MixinEntityVillager.java:255]: This should only be fired if `canVillagerPickupItem()` returns `true`.

**Review by JBYoshi — CHANGES_REQUESTED** [review]

**Comment by Zidane** [comment]
@Faithcaio 

Update this for @JBYoshi 's feedback.

File: src/main/java/org/spongepowered/common/mixin/core/entity/passive/MixinEntityVillager.java (hunk 1 above)
Comment (gabizou) [reply src/main/java/org/spongepowered/common/mixin/core/entity/passive/MixinEntityVillager.java:255]: Likewise, please javadoc what this method was originally doing if possible for future maintenance (as may be noticed in #1882)

**Comment by gabizou** [comment]
This will need to be re-done and re-target `api-8`. Bleeding is no longer a valid branch.

**Pull request closed without merging**